src-java/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java

```diff
@@ -418,9 +395,36 @@ public void shutdown() {
      * @return {@link CompletableFuture} with operation result.
      */
     public CompletableFuture<Optional<PacketInOutStats>> getPacketInOutStats() {
-        GrpcResponseObserver<PacketInOutStats> observer = new GrpcResponseObserver<>(address, GET_PACKET_IN_OUT_STATS);
         log.info("Getting packet in out stats for switch {}", address);
-        stub.showStatsPacketInOut(ShowPacketInOutStats.newBuilder().build(), observer);
-        return observer.future.thenApply(responses -> responses.stream().findFirst());
+
+        GrpcResponseObserver<PacketInOutStats> observer = new GrpcResponseObserver<>(address, GET_PACKET_IN_OUT_STATS);
+        extendChain(
+                dummy -> stub.showStatsPacketInOut(ShowPacketInOutStats.newBuilder().build(), observer),
+                observer.future);
+        return observer.future
+                .thenApply(responses -> responses.stream().findFirst());
+    }
+
+    private synchronized void extendChain(Consumer<Void> action, CompletableFuture<?> operation) {
```

Comment:
Can we change `Consumer<Void>` to `Runnable` here?

Reply:
yes, replaced